js/views/attachment_view.js

```diff
@@ -110,14 +109,22 @@
       }
     },
     onClick() {
-      if (this.isImage()) {
-        this.lightBoxView = new Whisper.LightboxView({ model: this });
-        this.lightBoxView.render();
-        this.lightBoxView.$el.appendTo(this.el);
-        this.lightBoxView.$el.trigger('show');
-      } else {
+      if (!this.isImage()) {
         this.saveFile();
+        return;
       }
+
+      const props = {
+        imageURL: this.objectUrl,
+        onSave: () => this.saveFile(),
+        // implicit: `close`
+      };
+      this.lightboxView = new Whisper.ReactWrapperView({
+        Component: Signal.Components.Lightbox,
+        props,
+        onClose: () => Signal.Backbone.Views.Lightbox.hide(),
+      });
+      Signal.Backbone.Views.Lightbox.show(this.lightboxView.el);
```

Comment:
Totally agree in terms of trying to avoid DOM manipulation for the new world. That’s why I created `Signal.Backbone` namespace to highlight these bridge modules between the old world and new world. Over time, our goal will be to kill `Signal.Backbone` 🔪